packages/lesswrong/components/comments/CommentsItem/CommentsItem.tsx

```diff
@@ -200,6 +280,14 @@ export const CommentsItem = ({
   const [replyFormIsOpen, setReplyFormIsOpen] = useState(false);
   const [showEditState, setShowEditState] = useState(false);
   const [showParentState, setShowParentState] = useState(showParentDefault);
+  const [activePotentialComment, setActivePotentialComment] = React.useState<0|1|2>(0)
+  const potentialCommentBodies = seededShuffle([...(comment.doppelComments ?? []).slice(0, 2), comment], comment._id)
+    .map(possiblyDoppelComment => {
+      const isDoppel = 'commentId' in possiblyDoppelComment
+      const content = isDoppel ? possiblyDoppelComment.content : (possiblyDoppelComment.contents?.html ?? "")
+      const doppelId = isDoppel ? possiblyDoppelComment.commentId : null
+      return { content, doppelId }
+    })
```

Comment:
If I was writing this I'd probably pull this out into a function that accepts a `CommentsList | CommentsListWithParentMetadata | DoppelCommentsFragment` and returns a `PotentialCommentBody`, or something.